controllers/addon/monitoring_federation_reconciler_test.go

```diff
@@ -220,10 +301,24 @@ func TestEnsureMonitoringFederation_Adoption(t *testing.T) {
 				Return(nil).
 				Maybe()
 
-			rec := &monitoringFederationReconciler{
-				client: c,
-				scheme: testutil.NewTestSchemeWithAddonsv1alpha1(),
-			}
+			uncachedC.
+				On("Get",
+					testutil.IsContext,
+					mock.Anything,
+					mock.IsType(&corev1.Secret{}), mock.Anything).
+				Return(nil)
+			uncachedC.
+				On("Get",
+					testutil.IsContext,
+					mock.Anything,
+					mock.IsType(&corev1.Secret{}), mock.Anything).
+				Return(nil)
+
+			c.
+				On("Create", mock.Anything, mock.IsType(&corev1.Secret{}), mock.Anything).Return(nil).Maybe()
+
+			c.
+				On("Update", mock.Anything, mock.IsType(&corev1.Secret{}), mock.Anything).Return(nil).Maybe()
```

Comment:
Please put a comment why you have added maybes. If we call `mock.AssertExpectations` at the end of a test case and have maybes added for mock calls, the assert expectations statement becomes very brittle.

Reply:
tackle this with the use of conditions